narwhals/_arrow/series.py

```diff
@@ -294,6 +295,25 @@ def zip_with(self: Self, mask: Self, other: Self) -> Self:
             )
         )
 
+    def sample(
+        self: Self,
+        n: int | None = None,
+        fraction: float | None = None,
+        *,
+        with_replacement: bool = False,
+    ) -> Self:
+        np = get_numpy()
+        pc = get_pyarrow_compute()
+        ser = self._native_series
+        num_rows = len(self)
+
+        if n is None and fraction is not None:
+            n = int(num_rows * fraction)
+
+        idx = np.arange(0, num_rows)
+        mask = np.random.choice(idx, size=n, replace=with_replacement)
```

Comment:
I am wondering if there is a better way of doing this.

Disclaimer, the issue with creating and then using a boolean mask directly is being able to deal with replacements